### PR TITLE
Support backfilling unpartitioned tables and non-date query parameters

### PR DESCRIPTION
**Pull request by ANich**
Closes https://github.com/mozilla/bigquery-etl/issues/4532

### CODE REVIEW

File: bigquery_etl/cli/query.py [hunk 1]
```diff
@@ -493,6 +493,14 @@ def info(ctx, name, sql_dir, project_id, cost, last_updated):
         click.echo("")
 
 
+def _parse_parameter(parameter: str, param_date: str) -> str:
+    # TODO: Parse more complex parameters such as macro_ds_add
+    if "{{ds}}" not in parameter:
```
Comment (ANich) [inline bigquery_etl/cli/query.py:498]: This should ideally use the same parsing logic as airflow does, as an import or something

File: bigquery_etl/cli/query.py [hunk 2]
```diff
@@ -740,17 +769,14 @@ def backfill(
...
+        partitioning_type = None
+        if metadata.bigquery and metadata.bigquery.time_partitioning:
             partitioning_type = metadata.bigquery.time_partitioning.type
-        else:
-            partitioning_type = PartitionType.DAY
-            click.echo("Bigquery partitioning type not set. Using PartitionType.DAY")
 
         match partitioning_type:
-            case PartitionType.DAY:
+            case None | PartitionType.DAY:
```
Comment (ANich) [inline bigquery_etl/cli/query.py:779]: Even if the table isn't partitioned, most queries still run daily, so format the dates in the same way

File: bigquery_etl/cli/query.py [hunk 3]
```diff
@@ -493,6 +493,15 @@ def info(ctx, name, sql_dir, project_id, cost, last_updated):
         click.echo("")
 
 
+def _parse_parameter(parameter: str, param_date: str) -> str:
+    # TODO: Parse more complex parameters such as macro_ds_add
+    param_name, param_type, param_value = parameter.split(":")
+    if param_type == "DATE" and param_value != "{{ds}}":
+        raise ValueError(f"Unable to parse parameter {parameter}")
```
Comment (ANich) [inline bigquery_etl/cli/query.py:500]: This should ideally use the same parsing logic as airflow does, as an import or something

**Review by scholtzan — APPROVED** [review]

**Review by fbertsch — APPROVED** [review]
LGTM. I'm a little concerned about the `_parse_parameter` function being a difficult-to-debug place, but since this is only used for ad-hoc purposes it should be fine to merge now and continue improving on.

Could you also update the `bqetl_backfill` DAG to support the new parameter?

File: bigquery_etl/cli/query.py [hunk 4]
```diff
@@ -493,6 +493,15 @@ def info(ctx, name, sql_dir, project_id, cost, last_updated):
         click.echo("")
 
 
+def _parse_parameter(parameter: str, param_date: str) -> str:
+    # TODO: Parse more complex parameters such as macro_ds_add
+    param_name, param_type, param_value = parameter.split(":")
+    if param_type == "DATE" and param_value != "{{ds}}":
```
[the remark this answers is not in the document]
Comment (ANich) [reply bigquery_etl/cli/query.py:499]: RE: this and top level comment, yeah this function should eventually be replaced by an Airflow util or a copy-paste of whatever Airflow's parameter parsing logic is

**Pull request merged**